### PR TITLE
Add  ml rfc doc

### PR DESCRIPTION
**Pull request by weicongs-amazon**
Signed-off-by:  Weicong Sun weicongs@amazon.com

### Description
Add rfc doc for ml project
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by zhanghg08 — APPROVED** [review]
Approved, thanks for the changes.

**Review by spbjss — APPROVED** [review]
looks good to me.

File: docs/ml-rfc.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    
+    Response: {
+     "taskId": "123", //the id of the job request
+     "status": "IN_PROGRESS" // the job execution status
+    
+    }
+   ```
+## Request for Comments:
+
+We would like comments and feedback on the proposal for building Machine Learning Framework here. Some specific questions we’re seeking feedback include
```
Comment (ylwu-amzn) [inline docs/ml-rfc.md:101]: Create one Github issue to gather customer's feedback and add the link in this doc?

Comment (weicongs-amazon) [reply docs/ml-rfc.md]: Added, thanks for the catch.

**Review by ylwu-amzn — APPROVED** [review]
LGTM.

**Pull request merged**